### DISCUSSION TITLE
Serverside Setting Cookies and Redirecting?

### DISCUSSION THREAD

**Discussion by muellerluke** [vercel/next.js#42913] [App Router]
Has anyone figured out a way to redirect and set cookies with NextJS 13 in the app directory? The cookies module is read only - although it says they're working on being able to set cookies as well. Is there no solution for the time being? Anyway we could get a timeframe for when this will be supported?

**Comment by dtpietrzak** [comment]
Check out the middleware docs for next12, they work for doing this. One side note, you have to have a pages directory or else it won't fire. So in my project I just have a blank pages dir.

**Reply by muellerluke** [reply]
Yeah my entire project is written in NextJS 12, but NextJS 13 has some amazing benefits with the app directory. However, I'm slowly starting to figure out that it's missing a ton of features that existed in the old version though.
